aspnet/dnx/projects.rst

```diff
@@ -1,8 +1,652 @@
+
+The DNX reads the lock file when running your application instead of rebuilding all the information that the DNU already generated. To understand the reason for that, imagine what the DNX has to do without the lock file:
+
+1. Find each dependency listed in the *project.json* file.
+2. Open the nuspec of each package and get all of their dependencies.
+3. Repeat step 2 for each dependency until it has the entire graph.
+4. Load all the dependencies.
+
+By using the lock file, this process is reduced to:
+
+1. Read the lock file.
+2. Load all the dependencies.
+
+There is significantly less disk IO involved in the second list.
+
+.. Note:: Because it has such a large impact on the cold start performance of DNX applications, the lock file is mandatory to run. If you do not have a lock file, the DNX will fail to load your application.
+
+In addition to this, the DNU also does things like determine if the package should be bread-crumbed for servicing and marks it as such in the lock file. See :doc:`Servicing DNX Applications <servicing>`.
+
+The lock file ensures that after you run ``dnu restore``, you have a fixed set of packages that you are referencing. When restoring, the DNU generates the lock file which specifies the exact versions that your project will use. This way, versions only get modified when you run ``dnu restore``, not during run-time. Restoring also ends up improving performance at run-time since the DNX no longer has to probe the packages directory to find the right version to use, the DNX just does what the lock file instructs the DNX to do. 
+
+There is a field in the lock file, ``locked``, which can be set to true either manually or via ``dnu restore —lock``. Setting this field to ``true`` specifies that ``dnu restore`` will just download the versions specified in the lock file and will not do any dependency graph walking or version selection. You can run ``dnu restore --lock`` to generate a locked lock file. Future restores will not change your installed version, unless you use ``dnu restore --unlock`` to remove the lock. You could lock your lock file and check it in on a release branch to ensure that you always get the exact version you expect, but leave it unlocked ()and ignored by source control on development branch(es).
+
+Specifying an alternative package locations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can add ``nupkg`` (NuGet packages) and ``source`` packages (not on a NuGet feed) to a project. To specify the location of these packages you must include the path to the NuGet package or to a packages folder, such as::
+
+	 dnu packages add newPackage.1.0.0.nupkg c:\packageStore  
+	 
+For additional details, run the help command::
+
+	dnu packages add --help
+
+Using commands
+--------------
+
+A command is a named execution of a .NET entry point with specific arguments. You can define commands in your *project.json* file:
```

Comment:
Suggestion: "A command is an alias for a package to use as an entry point, and an initial set of arguments"